src/array.rs

```diff
@@ -16,19 +16,31 @@ impl<T> Default for Array<T> {
 }
 
 impl<T: RuntimeType> Array<T> {
+    /// Creates an empty array.
     pub fn new() -> Self {
         Self::default()
     }
 
-    pub fn clear(&mut self) {
-        // TODO: drop members, CoTastkMemFree, zero members
+    /// Returns `true` if the array is empty.
+    pub fn is_empty(&self) -> bool {
+        self.len == 0
     }
 
-    pub fn as_slice(&self) -> &[T] {
-        if self.data.is_null() {
-            return &[];
+    /// Returns the length of the array.
+    pub fn len(&self) -> usize {
+        self.len as usize
+    }
+
+    /// Clears the contents of the array.
+    pub fn clear(&mut self) {
+        if self.is_empty() {
+            return;
+        }
+
+        unsafe {
+            std::ptr::drop_in_place(&mut self[..]);
+            CoTaskMemFree(self.data as _);
```

Comment:
We need to null the pointer out, otherwise the following is UB:

```rust
my_array.clear();
let slice: &[T] = &*my_array;
```

Reply:
Fixed.